### PR TITLE
[Issue #174] Hide pagination component if no results are found

### LINKED ISSUE #174: [Task]: Hide pagination component if there are no search results
### Summary

Currently when you preform a search and there are no results, the pagination component will still show with a full 7 pages that you can click through which do nothing. This is really awkward behavior.

### Acceptance criteria

Hide the pagination component when search results are 0

### PR DESCRIPTION
**Pull request by btabaska**
## Summary
Fixes #174 

## Context for reviewers
Updated all unit tests for pagination component. They were broken and commented out previously.

### CODE REVIEW

**Comment by acouch** [comment]
@btabaska , great catch!

File: frontend/tests/components/search/SearchPagination.test.tsx [hunk 1]
```diff
@@ -1,29 +1,26 @@
...
-
+// Mock the useSearchParamUpdater hook
 jest.mock("src/hooks/useSearchParamUpdater", () => ({
   useSearchParamUpdater: () => ({
-    updateQueryParams: mockUpdateQueryParams,
+    updateQueryParams: jest.fn(),
   }),
 }));
 
-// import SearchPagination, {
```
Comment (acouch) [inline frontend/tests/components/search/SearchPagination.test.tsx:16]: 👏🏿

File: frontend/tests/components/search/SearchPagination.test.tsx [hunk 2]
```diff
@@ -1,29 +1,26 @@
...
-//   PaginationPosition,
-// } from "../../../src/components/search/SearchPagination";
-
-// import { render } from "@testing-library/react";
```
Comment (acouch) [inline frontend/tests/components/search/SearchPagination.test.tsx:20]: 👏🏿 

*for removing commented out code

File: frontend/src/components/search/SearchPagination.tsx [hunk 3]
```diff
@@ -35,24 +35,28 @@ export default function SearchPagination({
   const pages = total || Number(totalPages);
 
   const updatePage = (page: number) => {
+    console.log("here");
```
Comment (acouch) [inline frontend/src/components/search/SearchPagination.tsx]: Looks like this got left here: 
```suggestion
```

File: frontend/src/components/search/SearchPagination.tsx [hunk 4]
```diff
@@ -35,24 +35,28 @@ export default function SearchPagination({
...
     updateTotalPages(String(total));
     updateTotalResults(totalResults);
     updateQueryParams(String(page), "page", query, scroll);
+    console.log("here2");
```
Comment (acouch) [inline frontend/src/components/search/SearchPagination.tsx]: This one too
```suggestion
```

**Comment by acouch** [comment]
@btabaska I found that if I click on the pager first, it doesn't disappear. Maybe if the page it is zero it should check and remove existing pagers, or we could just ignore this as it is not likely to happen often.

https://github.com/user-attachments/assets/e67885e7-7fe1-4520-b10b-38fd03857657

File: frontend/src/components/search/SearchPagination.tsx (hunk 3 above)
Comment (btabaska) [reply frontend/src/components/search/SearchPagination.tsx]: Great catch. This should have been caught by linting. Filing a ticket to add no-console-statement to Eslint config

File: frontend/src/components/search/SearchPagination.tsx (hunk 4 above)
Comment (btabaska) [reply frontend/src/components/search/SearchPagination.tsx]: Great catch, created. https://github.com/navapbc/simpler-grants-gov/issues/181 to add no-console to eslint

**Review by acouch — APPROVED** [review]

**Pull request merged**